### DISCUSSION TITLE
Cmdline completion documentation preview?

### DISCUSSION THREAD

**Discussion by tmillr** [hrsh7th/nvim-cmp#1671] [Ideas]
It would be really nice to have something like the `which-key` plugin, but for the cmdline instead of mappings. Or, in other words, when typing a cmd at the cmdline, after a configurable timeout (e.g. 2000 ms) a doc window/preview opens up showing nvim's help for the typed cmd. Are there certain settings or a plugin which accomplishes this? I couldn't find one. This would be helpful because alot of commands accept arguments, options, or bang, etc. and can have different behavior depending upon these factors. This could be opt-in and would be a helpful feature!

**Comment by Shougo** [comment]
Unfortunately, cmdline completion`getcompletion()` does not have the information.
It is hard.

For example, your command line manual completion `Tab` prints the information?
Because Vim/neovim has not.

**Reply by tmillr** [reply]
No neovim does not do this by default either. It might be somewhat difficult to implement but not impossible. All builtin cmdline/ex commands are documented and tagged (usually tagged under the command's name itself), and this plugin already has the capability of displaying docs in a floating window.

**Reply by Shougo** [reply]
Yes.  I think it is not possible.  Just hard.
